File: 1915. Number of Wonderful Substrings.go
func wonderfulSubstrings(word string) int64 {
	answer := 0
	freq := [1025]int{}
	freq[0] = 1
	curr := 0
	for _, w := range word {
		mask := 1 << (int(w) - int('a'))
		curr = curr ^ mask
		answer += freq[curr]
		for i := 0; i < 10; i++ {
			answer += freq[curr^(1<<i)]
		}
		freq[curr] += 1
	}
	return int64(answer)
}

File: 1915. Number of Wonderful Substrings.py
class Solution:
    def wonderfulSubstrings(self, word: str) -> int:
        curr = 0
        freq = [0]*1025
        freq[0] = 1
        answer = 0
        for w in word:
            mask = 1<<(ord(w)-ord('a'))
            curr = curr ^ mask
            answer += freq[curr]
            for i in range(10):
                answer += freq[curr ^ (1<<i)]
            freq[curr] += 1
        return answer